src/server/api/controllers/products.controller.js

```diff
@@ -2,72 +2,51 @@ const knex = require('../../config/db');
 const HttpError = require('../lib/utils/http-error');
 // const moment = require('moment-timezone');
 
-const getProducts = async (req) => {
-  let { limit, offset, sortKey, sortOrder } = req.query;
-  const allowedSortKeys = new Set(['name', 'price']);
-  const allowedSortOrders = new Set(['asc', 'desc']);
+const getProducts = async(req) => {
+    let { limit, offset, sortKey, sortOrder } = req.query;
+    const allowedSortKeys = new Set(['name', 'price']);
+    const allowedSortOrders = new Set(['asc', 'desc']);
 
-  limit = limit || 10;
+    limit = limit || 10;
 
-  offset = offset || 0;
+    offset = offset || 0;
 
-  sortKey = sortKey || 'name';
+    sortKey = sortKey || 'name';
 
-  sortOrder = sortOrder || 'asc';
+    sortOrder = sortOrder || 'asc';
 
-  const invalidParams =
-    isNaN(limit) ||
-    limit <= 0 ||
-    isNaN(offset) ||
-    offset < 0 ||
-    !allowedSortKeys.has(sortKey) ||
-    !allowedSortOrders.has(sortOrder);
+    const invalidParams =
+        isNaN(limit) ||
+        limit <= 0 ||
+        isNaN(offset) ||
+        offset < 0 ||
+        !allowedSortKeys.has(sortKey) ||
+        !allowedSortOrders.has(sortOrder);
 
-  if (invalidParams) {
-    throw new HttpError('Type or value of parameters is incorrect', 400);
-  }
+    if (invalidParams) {
+        throw new HttpError('Type or value of parameters is incorrect', 400);
+    }
 
-  try {
-    const totalCount = (
-      await knex('products').count('id', { as: 'count' }).first()
-    ).count;
-    const products = await knex('products')
-      .limit(limit)
-      .offset(offset)
-      .orderBy(sortKey, sortOrder);
+    try {
+        const totalCount = (
+            await knex('products').count('id', { as: 'count' }).first()
+        ).count;
+        const products = await knex('products')
+            .limit(limit)
+            .offset(offset)
+            .orderBy(sortKey, sortOrder);
 
-    return {
-      totalCount,
-      items: products,
-    };
-  } catch (error) {
-    return error.message;
-  }
+        return {
+            totalCount,
+            items: products,
+        };
+    } catch (error) {
+        return error.message;
+    }
 };
 
-const getDiscountProducts = async () => knex('products')
+const getDiscountProducts = async() => knex('products')
     .select(
-      'id',
-      'name',
-      'price',
-      'color',
-      'size',
-      'is_available',
-      'stock_amount',
-      'is_on_discount',
-      'discount_percent',
-      'picture',
-    )
-    .where('products.is_on_discount', '=', '1');
```

Comment:
This function getDiscountProducts looks a little incomplete. It doesn't have any error handeling. Did you deleted something?

Reply:
It seems that is it, I didn't change anything. Please check that on the swagger and if you think there is something to improve, please create an issue. So we can work on that.